lua/nvim-web-devicons.lua

```diff
@@ -1857,8 +1857,29 @@ local function get_default_icon()
   return default_icon
 end
 
+local function get_icon_by_extension(name, ext, opts)
```

Comment:
I _think_ this can be added to the API.

Please do so if appropriate, updating the readme.

Reply:
@alex-courtis This one, I am actually not very sure. `get_icon_by_extension` will ignore `icon_by_filename` so filename like `docker-compose.yaml` will be treated as just `yaml`.
I am not sure how much useful this is.

Still if you still think this is appropriate to add to the API, then I'll update it.

Reply:
That's fair enough. It doesn't really add anything but confusion.

The users can achieve everything via current API.